novawallet/Common/Services/RemoteSubscription/Substrate/Multisig/MultisigPendingOperationsSyncService.swift

```diff
@@ -0,0 +1,189 @@
+import Foundation
+import SubstrateSdk
+import Operation_iOS
+
+protocol MultisigPendingOperationsSyncServiceProtocol: ApplicationServiceProtocol {}
+
+class MultisigPendingOperationsSyncService {
+    let walletListLocalSubscriptionFactory: WalletListLocalSubscriptionFactoryProtocol
+
+    private let mutex = NSLock()
+
+    private let chainRepository: AnyDataProviderRepository<ChainModel>
+    private let callDataSyncService: MultisigCallDataSyncServiceProtocol
+    private let chainSyncServiceFactory: PendingMultisigChainSyncServiceFactoryProtocol
+    private let operationQueue: OperationQueue
+    private let workingQueue: DispatchQueue
+    private let logger: LoggerProtocol?
+
+    private var pendingOperationsChainSyncServices: [ChainModel.Id: PendingMultisigChainSyncServiceProtocol] = [:]
+
+    private var selectedMetaAccountProvider: StreamableProvider<ManagedMetaAccountModel>?
+
+    private var knownCallData: [Multisig.PendingOperation.Key: JSON] = [:]
+
+    private var selectedMetaAccount: MetaAccountModel? {
+        didSet {
+            if let selectedMetaAccount,
+               selectedMetaAccount != oldValue {
+                if selectedMetaAccount.multisigAccount != nil {
+                    createChainSyncServices(for: selectedMetaAccount)
+                } else {
+                    pendingOperationsChainSyncServices.forEach { $0.value.stopSyncUp() }
+                    pendingOperationsChainSyncServices = [:]
+                }
+            }
+        }
+    }
+
+    init(
+        chainRepository: AnyDataProviderRepository<ChainModel>,
+        callDataSyncService: MultisigCallDataSyncServiceProtocol,
+        chainSyncServiceFactory: PendingMultisigChainSyncServiceFactoryProtocol,
+        walletListLocalSubscriptionFactory: WalletListLocalSubscriptionFactoryProtocol,
+        operationQueue: OperationQueue,
+        workingQueue: DispatchQueue = DispatchQueue(label: "com.nova.wallet.pending.multisigs.sync.service"),
+        logger: LoggerProtocol? = nil
+    ) {
+        self.chainRepository = chainRepository
+        self.callDataSyncService = callDataSyncService
+        self.chainSyncServiceFactory = chainSyncServiceFactory
+        self.walletListLocalSubscriptionFactory = walletListLocalSubscriptionFactory
+        self.operationQueue = operationQueue
+        self.workingQueue = workingQueue
+        self.logger = logger
+    }
+}
+
+// MARK: - Private
+
+private extension MultisigPendingOperationsSyncService {
+    func performSetup() {
+        mutex.lock()
+        defer { mutex.unlock() }
+
+        selectedMetaAccountProvider = subscribeSelectedWalletProvider()
```

Comment:
I think we should use approach with the selected wallet passing from the SeviceCoordinator